src/lib/Microsoft.Health.Fhir.Ingest/Host/DeviceDataNormalizationExtensions.cs

```diff
@@ -56,11 +59,24 @@ private static NormalizationExceptionTelemetryProcessor TelemetryProcessorFactor
         {
             var options = serviceProvider.GetRequiredService<IOptions<NormalizationServiceOptions>>();
 
-            return options.Value.ErrorHandlingPolicy switch
+            var errorMessageService = serviceProvider.GetService<IErrorMessageService>();
+
+            if (errorMessageService == null)

```

Comment:
I'm not sure you need this switch, or even the alternate (zero parameter) constructor for NormalizationExceptionTelemetryProcessor. NormalizationExceptionTelemetryProcessor internally handles the case where errorMessageService is null. I think it makes sense to keep the logic that handles null errorMessageService in the consuming class, and simplify this code to not check if errorMessageService is null.

Reply:
the previous code as-is required the error message service to be passed in because we were new'ing the class, but I've refactored things a bit so we can use dependency injection and not "new" the class, so the null check is removed now